tests/ctst/steps/azureArchive.ts

```diff
@@ -407,11 +443,80 @@ Then('object {string} should expire in {int} days', async function (this: Zenko,
     assert(diff >= realTimeDays && diff < realTimeDays + 0.005);
 });
 
+Given('an azure archive location {string}', async function (this: Zenko, locationName: string) {
+    const locationConfig = {
+        name: locationName,
+        locationType: 'location-azure-archive-v1',
+        details: {
+            endpoint: AZURE_STORAGE_BLOB_URL,
+            bucketName: this.parameters.AzureArchiveContainer,
+            queue: {
+                type: 'location-azure-storage-queue-v1',
+                queueName: this.parameters.AzureArchiveQueue,
+                endpoint: AZURE_STORAGE_QUEUE_URL,
+            },
+            auth: {
+                type: 'location-azure-shared-key',
+                accountName: this.parameters.AzureAccountName,
+                accountKey: this.parameters.AzureAccountKey,
+            },
+        },
+    };
+    const result = await this.managementAPIRequest('POST', `/config/${this.parameters.InstanceID}/location`, {},
+        locationConfig);
+    assert.strictEqual(result.statusCode, 201);
+    this.addToSaved('locationName', locationName);
+    await Utils.sleep(60000); // Wait for location to be updated
+});
+
+When('i change azure archive location {string} container target', async function (this: Zenko, locationName: string) {
+    const result = await this.managementAPIRequest('GET', `/config/overlay/view/${this.parameters.InstanceID}`);
+    if ('err' in result) {
+        assert.ifError(result.err);
```

Comment:
This looks strange, if we detect an error, then shouldn't just we call the assert, that will prevent the rest of the test from running, allowing us to remove the root if condition?

Reply:
'err' field is not always in the result object, and `data` field is not here if `err` field is, so we need to check that to be able to use data as